config/Routes.cfm

```diff
@@ -29,18 +29,22 @@ limitations under the License.
 	// Sets automatic route extension detection and places the extension in the rc.format variable
 	// setExtensionDetection(true);
 	// The valid extensions this interceptor will detect
-	setValidExtensions('xml,json,jsont,rss,html,htm,cfm,print,pdf,doc');
+	setValidExtensions('xml,json,jsont,rss,html,htm,cfm,print,pdf,doc,txt');
 	// If enabled, the interceptor will throw a 406 exception that an invalid format was detected or just ignore it
 	// setThrowOnInvalidExtension(true);
 
 	// TO ENABLE FULL REWRITES REMOVE THE "INDEX.CFM" FROM THE LINES BELOW
-
+	if( len( getSetting( 'rewrites', false, '' ) ) && getSetting( 'rewrites', false, '' ) == true ){
+		rewriteURL = "";
+	} else {
+		rewriteURL = "index.cfm";
+	}
 	// Base URL
 	if( len(getSetting('AppMapping') ) lte 1){
-		setBaseURL("http://#cgi.HTTP_HOST##getContextRoot()#/index.cfm");
+		setBaseURL("http://#cgi.HTTP_HOST##getContextRoot()#/#rewriteURL#");
 	}
 	else{
-		setBaseURL("http://#cgi.HTTP_HOST##getContextRoot()#/#getSetting('AppMapping')#/index.cfm");
+		setBaseURL("http://#cgi.HTTP_HOST##getContextRoot()#/#getSetting('AppMapping')#/#rewriteURL#");
```

Comment:
What do you think about using this approach in the routing file for including or excluding the index.cfm?
This is a setting. 
We could set it using environment settings,